.github/workflows/samples-rust.yaml

```diff
@@ -0,0 +1,40 @@
+name: Samples Rust
+
+on:
+  push:
+    paths:
+      - 'samples/client/petstore/rust/**'
+      - 'samples/server/petstore/rust-server/**'
+  pull_request:
+    paths:
+      - 'samples/client/petstore/rust/**'
+      - 'samples/server/petstore/rust-server/**'
+
+jobs:
+  build:
+    name: Build Rust
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        sample:
+          - samples/client/petstore/rust/hyper/petstore
+          - samples/client/petstore/rust/reqwest/petstore-async-middleware
+          - samples/client/petstore/rust/reqwest/petstore-async
+          - samples/client/petstore/rust/reqwest/petstore-awsv4signature
+          - samples/client/petstore/rust/reqwest/petstore
+          - samples/server/petstore/rust-server/output/multipart-v3
+          - samples/server/petstore/rust-server/output/no-example-v3
+          - samples/server/petstore/rust-server/output/openapi-v3
+          - samples/server/petstore/rust-server/output/ops-v3
+          - samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing
+          - samples/server/petstore/rust-server/output/ping-bearer-auth
+          - samples/server/petstore/rust-server/output/rust-server-test
```

Comment:
Can this not just be replaced with?
```
samples/client/petstore/rust
samples/server/petstore/rust-server
```

Reply:
These are values of the variables `${{ matrix.sample }}`, which is used in the "working-directory" below in `Build`.

Reply:
Yeah I know, but I'm saying you don't need to manually go into each directory. You should just be able to run the build command in the workspace root (see my other comment)

Reply:
Ah I see what you mean now. You're right. We can simply use the 2 lines you mentioned above.